workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/SpringBdkGateway.java

```diff
@@ -16,15 +18,17 @@
   private final StreamService streamService;
   private final UserService userService;
   private final ConnectionService connectionService;
+  private final SymphonyGroupService groupService;
 
   @Autowired
   public SpringBdkGateway(MessageService messageService,
       StreamService streamService, UserService userService,
-      ConnectionService connectionService) {
+      ConnectionService connectionService, @Lazy SymphonyGroupService groupService) {
```

Comment:
For my knowledge, why groupService bean is not created at the same time as the other ones (streamService..)?

Reply:
It comes from https://github.com/finos/symphony-bdk-java/blob/main/docs/extension.md#access-your-extensions-service-in-spring-boot

and from https://github.com/finos/symphony-bdk-java/blob/e4c7ccc3714c1efa1b028a0aebfa00c56353dda6/symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/config/BdkExtensionConfig.java#L56

where we create and add the beans manually

with lazy we have this part executed first and then the use of the SymphonyGroupService bean is resolved later (and then it is available in the spring context)